internal/include/otel_types.h

```diff
@@ -0,0 +1,145 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef _OTEL_TYPES_H
+#define _OTEL_TYPES_H
+
+#include "go_types.h"
+#include "common.h"
+
+/* Defintions should mimic structs defined in go.opentelemetry.io/otel/attribute */
+typedef u64 attr_val_type_t;
+
+#define INVALID 0
+#define BOOL 1
+#define INT64 2
+#define FLOAT64 3
+#define STRING 4
+#define BOOLSLICE 5
+#define INT64SLICE 6
+#define FLOAT64SLICE 7
+#define STRINGSLICE 8
+
+typedef struct go_otel_attr_value {
+	attr_val_type_t  vtype;
+	u64              numeric;
+	struct go_string string;
+	struct go_iface	 slice;
+} go_otel_attr_value_t;
+
+typedef struct go_otel_key_value {
+	struct go_string     key;
+	go_otel_attr_value_t value;
+} go_otel_key_value_t;
+
+#define OTEL_ATTRIBUTE_KEY_MAX_LEN      (32)
+#define OTEL_ATTRIBUTE_VALUE_MAX_LEN    (128)
+#define OTEL_ATTRUBUTE_MAX_COUNT        (16)
```

Comment:
These seem small (the default limits are higher than some of these https://github.com/open-telemetry/opentelemetry-go/blob/204be6157f13594eb92a91914aaa7a7561bfd08a/sdk/trace/span_limits.go#L26). Is there any way we can support more dynamic string lengths?

Reply:
There is but it was challenging to make the eBPF verifier happy with more efficient encoding. I initially tried to pack all the string keys and values into the same buffer and separate them with null bytes. This will create a limit on the combined size of the strings instead of the length of each one and will allow more flexible string lengths. 
We can make the `OTEL_ATTRUBUTE_MAX_COUNT` bigger, but it will make the span entry sent to the perf buffer larger (few KB).